Lib/test/test_concurrent_futures.py

```diff
@@ -238,6 +238,29 @@ def test_thread_names_default(self):
             self.assertRegex(t.name, r'ThreadPoolExecutor-\d+_[0-4]$')
             t.join()
 
+    def test_default_max_queue_size(self):
+        with futures.ThreadPoolExecutor() as executor:
+            self.assertEqual(executor._work_queue.maxsize, 0)
+
+    def test_custom_max_queue_size(self):
+        qsize = 1
+        with futures.ThreadPoolExecutor(max_queue_size=qsize) as executor:
+            # test custom queue size was passed down
+            self.assertEqual(executor._work_queue.maxsize, qsize)
+
+            # test executor works with custom size
+            n = 10
+
+            def process_item(item):
+                return item + 1
```

Comment:
Perhaps add a small sleep() here to help the queue reach its limit at some point?